x/twap/strategy.go

```diff
@@ -0,0 +1,40 @@
+package twap
+
+import (
+	"time"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/osmosis-labs/osmosis/v13/x/twap/types"
+)
+
+type twapStrategy interface {
+	getTwapToNow(
+		ctx sdk.Context,
+		poolId uint64,
+		baseAssetDenom string,
+		quoteAssetDenom string,
+		startTime time.Time) (sdk.Dec, error)
+	computeTwap(startRecord types.TwapRecord, endRecord types.TwapRecord, quoteAsset string) (sdk.Dec, error)
+}
+
+type arithmetic struct {
+	keeper Keeper
+}
+
+var _ twapStrategy = &arithmetic{}
+
+func (s *arithmetic) getTwapToNow(
+	ctx sdk.Context,
+	poolId uint64,
+	baseAssetDenom string,
+	quoteAssetDenom string,
+	startTime time.Time,
+) (sdk.Dec, error) {
+	// decide whether we want to use arithmetic or geometric twap
```

Comment:
What do you mean by "decide between arithmetic or geometric" here? I think this might confuse folks.

`arithmetic` strategy should always utilize the respective arithmetic methods

Reply:
isn't the idea to have a `isArithmeticType` bool to decide if we want to calculate arithmetic or geometric in this function?

Reply:
There is no `isArithmeticType` boolean in this function.

If you are referring to some changes introduced in: https://github.com/osmosis-labs/osmosis/pull/3420, those would actually need to be refactored to be integrated with the strategy pattern here. So the boolean parameter `isArithmeticTwap` in #3420 would likely be changed to taking a strategy instead.

I suggest assuming that #3420 does not exist for the context of this PR. We will integrate them after one or the other is merged

Reply:
aah gotchu